go/cmd/dolt/commands/engine/sqlengine.go

```diff
@@ -278,6 +286,67 @@ func configureBinlogReplicaController(config *SqlEngineConfig, engine *gms.Engin
 	return nil
 }
 
+// configureEventScheduler configures the event scheduler with the |engine|.
+func configureEventScheduler(config *SqlEngineConfig, engine *gms.Engine, sessFactory sessionFactory, pro dsqle.DoltDatabaseProvider) error {
+	// need to give correct user, use the definer as user to run the event definition queries
+	ctxFactory := sqlContextFactory()
+
+	// getCtxFunc is used to create new session context for event scheduler.
+	// It starts a transaction that needs to be committed using the function returned.
+	getCtxFunc := func() (*sql.Context, func() error, error) {
+		sess, err := sessFactory(sql.NewBaseSession(), pro)
+		if err != nil {
+			return nil, func() error { return nil }, err
+		}
+
+		newCtx, err := ctxFactory(context.Background(), sess)
+		if err != nil {
+			return nil, func() error { return nil }, err
+		}
+
+		// Starting transaction at the start of the server to load all events to
+		// the event scheduler is causing read replica test to fail as it tries
+		// pulling from remotes for branches that its remoteRef does not exist yet.
+		err = sql.SystemVariables.SetGlobal(dsess.SkipReplicationErrors, true)
+		if err != nil {
+			return nil, func() error { return nil }, err
+		}
```

Comment:
I ended up changing the replication test, so that it created the `feature` branch first and then it updated its replication configuration to specify that branch as being replicated. That was enough to get the test to pass without this block.